cloudinit/sources/DataSourceOracle.py

```diff
@@ -297,14 +297,20 @@ def _add_network_config_from_opc_imds(self, set_primary: bool = False):
                 )
                 continue
             name = interfaces_by_mac[mac_address]
-            prefix = ipaddress.ip_network(
-                vnic_dict["subnetCidrBlock"]
-            ).prefixlen
+            network = ipaddress.ip_network(vnic_dict["subnetCidrBlock"])
+            gateway = vnic_dict["virtualRouterIp"]
 
             if self._network_config["version"] == 1:
                 subnet = {
                     "type": "static",
-                    "address": f"{vnic_dict['privateIp']}/{prefix}",
+                    "address": f"{vnic_dict['privateIp']}/{network.prefixlen}",
+                    "routes": [
+                        {
+                            "network": str(network.network_address),
+                            "prefix": network.prefixlen,
+                            "gateway": gateway,
```

Comment:
Same "metric': 100 question as below since this is the default route on the primary nic

Reply:
work item for future.